pyrealm/pmodel/subdaily.py

```diff
@@ -322,6 +323,14 @@ def __init__(
             1 / calc_ftemp_arrh(tk_acclim, self.env.pmodel_const.subdaily_jmax25_ha)
         )
 
+        """Instantaneous optimal :math:`x_{i}`, :math:`V_{cmax}` and :math:`J_{max}`"""
+        if init_realised is not None:
+            self.init_xi_real, self.init_vcmax_real, self.init_jmax_real = init_realised
+        else:
+            self.init_xi_real = self.pmodel_acclim.optchi.xi
+            self.init_vcmax_real = self.vcmax25_opt
+            self.init_jmax_real = self.jmax25_opt
+
         # 5) Calculate the realised daily values from the instantaneous optimal values
         self.xi_real: NDArray = memory_effect(
             self.pmodel_acclim.optchi.xi, alpha=alpha, allow_holdover=allow_holdover
```

Comment:
The init values need to be passed into `memory_effect` - they should replace the initial assumption that $R_{0} = O_{0}$ at $t=0$. So I think what is needed is an `initial_values: NDArray | None = None` argument to `memory_effect` and then the line `memory_values[0] = values[0]` can conditionally change to `memory_values[0] = initial_values`.

A sensible test for this would be to take the existing test and then split it into two time chunks - feeding the last realised value from the first chunk into the initial values of the second chunk should then duplicate the results of an all in one run.

Reply:
Attempted to implement in https://github.com/ImperialCollegeLondon/pyrealm/pull/190/commits/9e3d36d326bbcdfdd5aee00218e5b72877f2b8fa

Reply:
There is a mismatch in dimensions when assigning the values in `memory_effect` : `values` , `initial_values`  (as per your suggestion) and `memory_values`  have the same dimension. So assigning `memory_values[0] = initial_values` fails. Should it be `memory_values[0] = initial_values[0]` ? And should all the remaining assignments for `memory_values` stay as they are (following `values`, or go to `initial_values`?

Reply:
The arrays of data in `subdaily` can have multiple dimensions but the first axis must be time. When users are providing initial values, those need to be a single set of observations for the first time point, so should have the shape of a slice along that time axis. I think when that is enforced, the original assignment is correct. 

So the input checking for user values should be:

```python
In [59]: data = np.arange(12).reshape((3,4))
In [60]: data
Out[60]: 
array([[ 0,  1,  2,  3],
       [ 4,  5,  6,  7],
       [ 8,  9, 10, 11]])
In [61]: init = np.arange(4)
In [62]: init
Out[62]: array([0, 1, 2, 3])
In [63]: data[0].shape == init.shape
Out[63]: True
```

Reply:
After our discussion I corrected that bit back, and adapted the assignment bit as shown in https://github.com/ImperialCollegeLondon/pyrealm/pull/190#discussion_r1786448078, in this commit: https://github.com/ImperialCollegeLondon/pyrealm/pull/190/commits/504ad7410f6bedb7d591289c712f7be650826663 .

Reply:
I implemented a test following your suggestion as follows: 


```
def test_memory_effect_chunked(inputs_whole, alpha):
    from pyrealm.pmodel import memory_effect

    result_whole = memory_effect(inputs_whole, alpha=alpha)

    [inputs_chunk1, inputs_chunk2] = np.split(inputs_whole, [5], axis=0)

    result_chunk1 = memory_effect(inputs_chunk1, alpha=alpha)

    result_chunk2 = memory_effect(
        inputs_chunk2, initial_values=result_chunk1[-1], alpha=alpha
    )

    assert np.allclose(result_whole[-1], result_chunk2[-1])
```

However, `np.allclose` fails -- the values are "kind of close", but different. For example, for the 2D test, it compares `array([ 8.001953,  9.001953, 10.001953, 11.001953], dtype=float32)` and `array([ 7.9414062,  8.941406 ,  9.941406 , 10.941406 ], dtype=float32))` .

Reply:
From @davidorme : 

"If we do this:
```
    memory_values = np.empty_like(values, dtype=np.float32)
    if initial_values is None:
        memory_values[0] = values[0]
    else:
        memory_values[0] = initial_values * (1 - alpha) + values[0] * alpha
```
Then the current test works as expected. I think that’s a better API. People might want to feed in a year at a time for example, and having to get the overlapping observation is finicky. We might then want to rename initial_values to previous_values or something similar to make this clear. "